sky/clouds/vast.py

```diff
@@ -234,6 +249,17 @@ def _make(instance_list):
                 resource_list.append(r)
             return resource_list
 
+        # Resolve datacenter_only config first (used for all instance filtering)
+        datacenter_only = skypilot_config.get_nested(
+            ('vast', 'datacenter_only'),
+            None,
+            override_configs=resources.cluster_config_overrides)
+        if datacenter_only is None:
+            datacenter_only = skypilot_config.get_nested(
+                ('vast', 'secure_only'),
+                False,
+                override_configs=resources.cluster_config_overrides)
```

Comment:
How about we just keep one of them? I don't think we need a strong backward compatibility here : )